### PR TITLE
Install sail-operator 0.1.0

### LINKED ISSUE #401: Installing sail operator without kustomize
Follows up from #400.

The sail operator moved away from using `kustomize` to `helm` to install locally which means we can no longer install the same way for local dev / testing (https://github.com/maistra/istio-operator/pull/1576). This leaves `ISTIO_INSTALL_SAIL=true` non-functional (edit: this has been fixed by pinning the sail version to the commit before the manifests were removed).

Currently there isn't a helm package/index to consume from in the GitHub repo https://github.com/maistra/istio-operator/tree/maistra-3.0/chart so it does not look like we can easily install from the helm chart.

### PR DESCRIPTION
**Pull request by adam-cattermole**
Installs sail-operator using helm from [istio-ecosystem/sail-operator@0.1.0](https://github.com/istio-ecosystem/sail-operator/releases/tag/0.1.0) and re-enables integration tests.

Requires #893, #952
Continues #696
Closes #401

### CODE REVIEW

**Comment by adam-cattermole** [comment]
I have tested on top of RLP from #893 and can see things working as expected.

File: config/dependencies/istio/sail/istio.yaml [hunk 1]
```diff
@@ -1,15 +1,12 @@
-apiVersion: operator.istio.io/v1alpha1
+apiVersion: sailoperator.io/v1alpha1
 kind: Istio
 metadata:
   name: default
 spec:
-  version: v1.20.0
+  # Supported values for sail-operator v0.1.0 are [v1.22.4,v1.23.0]
+  version: v1.23.0
```
[the remark this answers is not in the document]
Comment (maleck13) [reply config/dependencies/istio/sail/istio.yaml:7]: What version of GWAPI does that align with?

**Comment by maleck13** [comment]
changes look good. I will verify it here locally also before approving

File: config/dependencies/istio/sail/istio.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (adam-cattermole) [reply config/dependencies/istio/sail/istio.yaml:7]: I believe v1.23 is still GWAPI v1.1 and istio v1.24 supports GWAPI v1.2

**Comment by adam-cattermole** [comment]
I've tested this locally on top of the changes from #952 and the auth side of things is working

**Comment by maleck13** [comment]
@adam-cattermole is there anything that needs to be done with the quick start? Still haven't got to try this out yet have a backlog of stuff at the moment. Happy to approve though as you seem to have done a fair bit of validation here

**Comment by adam-cattermole** [comment]
No worries, I'll have a go with the quickstart and report back, but either way I think I'll wait on merge until #952 is in

**Comment by adam-cattermole** [comment]
@maleck13 Made changes for the quickstart - default is now sail and there was a couple changes to install this new version, including dependence on helm installed. I've got a PR to update the singe cluster install guide here https://github.com/Kuadrant/docs.kuadrant.io/pull/144

**Comment by maleck13** [comment]
awsome will try this now

**Review by maleck13 — APPROVED** [review]
Looks good. Worked well locally also.

**Pull request merged**